lib/maths/unittest/CBoostedTreeTest.cc

```diff
@@ -1221,6 +1221,73 @@ BOOST_AUTO_TEST_CASE(testEstimateMemoryUsedByTrain) {
     }
 }
 
+BOOST_AUTO_TEST_CASE(testEstimateMemoryUsedByTrainWithTestRows) {
+
+    // Test estimation of the memory used training a model.
+
+    test::CRandomNumbers rng;
+
+    std::size_t rows{1000};
+    std::size_t cols{6};
+    std::size_t capacity{600};
+
+    for (std::size_t test = 0; test < 3; ++test) {
+        TDoubleVecVec x(cols - 1);
+        std::size_t num_test_rows{((test + 1) * 100)};
```

Comment:
nit: can we stick with camel case
```suggestion
        std::size_t numTestRows{((test + 1) * 100)};
```